Newer kernels complain about a void function returning NULL and a discarding of const:

| netana_enum.c:251:21: warning: assignment discards 'const' qualifier from pointer target type [-Wdiscarded-qualifiers]
|   251 |                 buf = of_get_property(pdev->dev.of_node, "startuptype", &buflen);
|       |                     ^

|                  from OS_Linux.c:37:
| OS_Linux.c: In function 'OS_WritePCIConfig':
| kernel-source/include/linux/stddef.h:8:14: error: 'return' with a value, in function returning void [-Werror=return-type]
|     8 | #define NULL ((void *)0)
|       |              ^
| OS_Linux.c:239:24: note: in expansion of macro 'NULL'
|   239 |                 return NULL;
|       |                        ^~~~

diff -aurN a/netana_enum.c b/netana_enum.c
--- a/netana_enum.c	2021-01-21 12:18:19.750230947 +0100
+++ b/netana_enum.c	2021-01-27 06:53:11.149678255 +0100
@@ -246,7 +246,7 @@
                 ret = -EINVAL;
                 goto err;
         } else {
-                char *buf;
+                const char *buf;
                 int buflen;
                 buf = of_get_property(pdev->dev.of_node, "startuptype", &buflen);
                 if (buf && !strncmp(buf, "flash", buflen)) {
diff -aurN a/OS_Linux.c b/OS_Linux.c
--- a/OS_Linux.c	2021-01-21 12:18:19.034217927 +0100
+++ b/OS_Linux.c	2021-01-27 06:53:46.498261831 +0100
@@ -234,10 +234,10 @@
         struct netana_info *ptDeviceInfo = (struct netana_info*)pvOSDependent;
 
         if(!pvOSDependent)
-                return NULL;
+                return;
 
         if(!ptDeviceInfo->pcidevice)
-                return NULL;
+                return;
 
         pci_ret = WritePCIConfig(ptDeviceInfo->pcidevice, 256, pvPCIConfig);
 
